### ISSUE TITLE
Message:  RPC Subscription stream should be infinite

### ISSUE THREAD

**Issue by nirnaeth-arnoediad** [availproject/avail-light#388]
I am running avail light client on ubuntu 22.04 amd64 architecture LC version is v1.7.4. My client crashed like this:

```
2023-12-15T10:33:49.633435Z  INFO avail_light::network: Cells fetched from DHT block_number=14448 cells_total=8 cells_fetched=0 cells_verified=0 fetch_elapsed=10.002151687s proof_verification_elapsed=203.551µs
2023-12-15T10:33:49.754989Z  INFO avail_light::network: Cells fetched from RPC block_number=14448 cells_total=8 cells_fetched=8 cells_verified=8 fetch_elapsed=110.676771ms proof_verification_elapsed=10.789979ms
2023-12-15T10:33:49.755207Z  INFO avail_light::sync_client: App index Ok(DataLookup { index: [(AppId(0), 0..1)] }) block_number=14449
2023-12-15T10:33:49.755222Z  INFO avail_light::sync_client: Synced block header block_number=14449 elapsed=16.831µs
2023-12-15T10:33:49.755257Z  INFO avail_light::api::v2: Message published to clients topic=ConfidenceAchieved published=0 failed=0
2023-12-15T10:33:50.851187Z  INFO avail_light::network: Cells fetched from DHT block_number=14449 cells_total=8 cells_fetched=0 cells_verified=0 fetch_elapsed=1.095730806s proof_verification_elapsed=213.59µs
2023-12-15T10:33:50.946020Z  INFO avail_light::network: Cells fetched from RPC block_number=14449 cells_total=8 cells_fetched=8 cells_verified=8 fetch_elapsed=83.210699ms proof_verification_elapsed=11.595919ms
2023-12-15T10:33:50.950271Z  INFO avail_light::sync_client: App index Ok(DataLookup { index: [(AppId(0), 0..12)] }) block_number=14450
2023-12-15T10:33:50.950282Z  INFO avail_light::sync_client: Synced block header block_number=14450 elapsed=12.393µs
2023-12-15T10:33:50.950312Z  INFO avail_light::api::v2: Message published to clients topic=ConfidenceAchieved published=0 failed=0
2023-12-15T10:33:50.958860Z  INFO avail_light::network::p2p::event_loop: Cell upload success rate for block 14448: 8/8. Duration: 1
2023-12-15T10:33:52.046668Z  INFO avail_light::network::p2p::event_loop: Cell upload success rate for block 14449: 8/8. Duration: 1
The application panicked (crashed).
Message:  RPC Subscription stream should be infinite
Location: /root/avail-light/src/network/rpc/event_loop.rs:217
```

My LC was syncing from block 0. I am adding the generated crash report. Hope this helps.
[report-ea65bb3d-8043-4b67-8227-2058fd94416f.toml.zip](https://github.com/availproject/avail-light/files/13685117/report-ea65bb3d-8043-4b67-8227-2058fd94416f.toml.zip)

To communicate if needed:
Discord: nirnaeth.
E-mail: nirnaeth@zohomail.eu

**Comment by nirnaeth-arnoediad** [comment]
hey @sh3ll3x3c I want to inform you that I faced the issue on v1.7.6 too. Here the logs:

`2023-12-15T10:33:49.633435Z  INFO avail_light::network: Cells fetched from DHT block_number=14448 cells_total=8 cells_fetched=0 cells_verified=0 fetch_elapsed=10.002151687s proof_verification_elapsed=203.551µs
2023-12-15T10:33:49.754989Z  INFO avail_light::network: Cells fetched from RPC block_number=14448 cells_total=8 cells_fetched=8 cells_verified=8 fetch_elapsed=110.676771ms proof_verification_elapsed=10.789979ms
2023-12-15T10:33:49.755207Z  INFO avail_light::sync_client: App index Ok(DataLookup { index: [(AppId(0), 0..1)] }) block_number=14449
2023-12-15T10:33:49.755222Z  INFO avail_light::sync_client: Synced block header block_number=14449 elapsed=16.831µs
2023-12-15T10:33:49.755257Z  INFO avail_light::api::v2: Message published to clients topic=ConfidenceAchieved published=0 failed=0
2023-12-15T10:33:50.851187Z  INFO avail_light::network: Cells fetched from DHT block_number=14449 cells_total=8 cells_fetched=0 cells_verified=0 fetch_elapsed=1.095730806s proof_verification_elapsed=213.59µs
2023-12-15T10:33:50.946020Z  INFO avail_light::network: Cells fetched from RPC block_number=14449 cells_total=8 cells_fetched=8 cells_verified=8 fetch_elapsed=83.210699ms proof_verification_elapsed=11.595919ms
2023-12-15T10:33:50.950271Z  INFO avail_light::sync_client: App index Ok(DataLookup { index: [(AppId(0), 0..12)] }) block_number=14450
2023-12-15T10:33:50.950282Z  INFO avail_light::sync_client: Synced block header block_number=14450 elapsed=12.393µs
2023-12-15T10:33:50.950312Z  INFO avail_light::api::v2: Message published to clients topic=ConfidenceAchieved published=0 failed=0
2023-12-15T10:33:50.958860Z  INFO avail_light::network::p2p::event_loop: Cell upload success rate for block 14448: 8/8. Duration: 1
2023-12-15T10:33:52.046668Z  INFO avail_light::network::p2p::event_loop: Cell upload success rate for block 14449: 8/8. Duration: 1
The application panicked (crashed).
Message:  RPC Subscription stream should be infinite
Location: /root/avail-light/src/network/rpc/event_loop.rs:217`

**Comment by sh3ll3x3c** [comment]
Hey @nirnaeth-arnoediad, thanks again for reporting.

In v1.7.6 we only implemented the first iteration of the retry mechanism in which we handle the initial node connection failures.

The yesterday released v1.7.7 rolled out the second iteration which handles the mid-execution drops, which you just saw. You can try it out, it should solve this problem and if you run across it again, please feel free to report to us - the feedback really means a lot to the team.
